File: funnyvote/app/src/main/java/com/heaton/funnyvote/FirstTimePref.java
package com.heaton.funnyvote;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by heaton on 2017/2/26.
 */
public class FirstTimePref {
    private static final String SP_FIRST_TIME = "first_time";
    public static final String SP_FIRST_MOCK_DATA = "first_mock_data";
    public static final String SP_FIRST_INTRODUCTION_PAGE = "first_introduction_page";
    public static final String SP_FIRST_INTRODUTCION_QUICK_POLL = "first_introduction_quick_poll";
    public static final String SP_FIRST_ENTER_UNPOLL_VOTE = "first_enter_unpoll_vote";
    SharedPreferences preferences;

    private static FirstTimePref instance;

    public static FirstTimePref getInstance(Context context) {
        if (instance == null) {
            instance = new FirstTimePref(context);
        }
        return instance;
    }

    private FirstTimePref(Context context) {
        preferences = context.getSharedPreferences(SP_FIRST_TIME,Context.MODE_PRIVATE);
    }
    public SharedPreferences getPreferences() {
        return preferences;
    }
}

File: funnyvote/app/src/main/java/com/heaton/funnyvote/FirstTimePref.kt
package com.heaton.funnyvote

import android.content.Context
import android.content.SharedPreferences

/**
 * Created by heaton on 2017/2/26.
 */
class FirstTimePref private constructor(context: Context) {
    var preferences: SharedPreferences
        internal set

    init {
        preferences = context.getSharedPreferences(SP_FIRST_TIME, Context.MODE_PRIVATE)
    }

    companion object {
        private val SP_FIRST_TIME = "first_time_use"
        val SP_FIRST_MOCK_DATA = "first_mock_data"
        val SP_FIRST_INTRODUCTION_PAGE = "first_introduction_page"
        val SP_FIRST_INTRODUTCION_QUICK_POLL = "first_introduction_quick_poll"
        val SP_FIRST_ENTER_UNPOLL_VOTE = "first_enter_unpoll_vote"

        private var instance: FirstTimePref? = null

        fun getInstance(context: Context): FirstTimePref {
            if (instance == null) {
                instance = FirstTimePref(context)
            }
            return instance as FirstTimePref
        }
    }
}
